aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -127,6 +127,31 @@ struct WelfordOps {
   }
 };
 
+template <typename acc_t, typename data_t>
+struct NanProdOps {
+  inline C10_DEVICE acc_t reduce(acc_t a, data_t b, int64_t /*idx*/) const {
+    return a * (at::_isnan(b) ? acc_t{1.} : acc_t{b});
```

Comment:
There are 4 cases here, right?

- a and b are both real
- a and b are both nan
- a is nan but b is real
- a is real but b is nan

Is that correct? 

What should happen in each of those cases? Also, what's the best way to optimize this for the case where both are real? That is, what is the fewest checks this can do before it computes (and returns) `a * b` if both a and b are real-valued?